### DISCUSSION TITLE
Welcome to Evennia Discussions!

### DISCUSSION THREAD

**Discussion by Griatch** [evennia/evennia#2267] [General Discussion]
## 👋 Welcome to the Evennia Discussions!

This is a central point for the Evennia community to communicate. Here you can ask questions, introduce yourself, talk about your Evennia project, discuss and interact with other users and developers! This complements the official Evennia support channel on [Discord](https://discord.gg/AJJpcRUhtF). When you make a post, the chat channel will be notified.

- If you find clear issues/bugs, don't report them here but in the [issue tracker](https://github.com/evennia/evennia/issues/new/choose).
- See [Code of Conduct](https://github.com/evennia/evennia/blob/ff2ab3a3a56eb2641b1f7eee65501049fa1dbcd9/CODE_OF_CONDUCT.md) for ways to behave. Basically, keep it civil.
- To get started, go and [introduce yourself in the Introductions category](https://github.com/evennia/evennia/discussions?discussions_q=category%3A%22Greetings+and+Introductions%22)!
- https://www.evennia.com is the central Evennia homepage.
- [Evennia Discord support channel](https://discord.gg/AJJpcRUhtF) to chat in our very active community.
- https://www.evennia.com/docs/latest holds the Evennia docs.
- (The [old Forum / mailing-list](https://groups.google.com/g/evennia) on Google groups is closed and acts as an archive of even older discussions).

## Using Markdown

Discussions uses _Markdown_ syntax with github extensions.

- [Markdown basic syntax](https://docs.github.com/en/free-pro-team@latest/github/writing-on-github/basic-writing-and-formatting-syntax)
- [HTML tags allowed in text](https://gist.github.com/coolaj86/89821fe046623d5503ce5c4133e70506)

<details>
<summary>Click here to expand the post with some Markdown usage examples.</summary>

#### Basic syntax

- _Italic text_ : `_Italic text_`
- **Bold text**: `**Bold text**`
- ~~Strikethrough text~~: `~~Strikethrough text~~`
- `verbatim text`: \`verbatim text\`
- [clickable link](https://www.evennia.com): `[clickable link](https://www.evennia.com)`
- Line separator: `----`
- Emojis are created by `:smile:` :smile: . Starting to write `:` will present a popup with the available emojis as you type.
- ![image text](https://github.githubassets.com/images/icons/emoji/unicode/1f4ac.png): `![image text](https://url/to/image.png)`. You can also drag&drop or click the bottom of the editor.

#### Headings

```
#  level 1 heading (biggest)
##  level 2 heading
### level 3 heading
#### level 4 heading
##### level 5 heading (smallest)
```

#### Lists

```
- List item one
- List item two 
```

- List item one
- List item two 

```
1. Numbered item one
2. Numbered item two
```

1. Numbered item one
2. Numbered item two

#### Verbatim multi-line text and Code blocks:

````
```
This is verbatim, multi-line text that retains 
the exact formatting
as you wrote ...                  ...  it!
```
````
```
This is verbatim, multi-line text that retains 
the exact formatting
as you wrote ...                  ...  it!
```

For code syntax highlighting, the language must be specified. [Here is a list of supported syntax-highlighting languages](https://github.com/jincheng9/markdown_supported_languages#heres-a-full-list-of-supported-languages).

````
```python
def hello_world(x, y):
    print(x, y)
```
````

```python
def hello_world(x, y):
    print(x, y)
```

````
```html
<div class="nice">
   This is a nice text.
</div>
```
````

```html
<div class="nice">
   This is a nice text.
</div>
```

#### Clickable, expanded section

The `<details>` HTML tag can be used for 'spoilers', hiding extra info and so on. This whole markdown-example section was created in this way. 

```markdown
Here is an example question with a hidden answer!
<details>
<summary>Click here to get the answer!</summary>
The answer is that you clicked correctly!
</details>
```

Here is an example question with a hidden answer! 
<details>
<summary>Click here to get the answer!</summary>
The answer is that you clicked correctly!
</details>

</details>

**Comment by Griatch** [comment]
Discussions are still in beta and is missing some features, for example:

- Posting a discussion message does not appear to trigger a webhook yet, so we can't notify the IRC/Discord channels at this point. _Update: This was added now!_
- The categories does not appear to be possible to sort arbitrarily - they only sort alphabetically

But still, seems to be working fine so far.

**Comment by kovitikus** [comment]
@Griatch you said to let you know of any feedback on the Discussions thing, so that you can pass it along to Github.

I don't see any support for adding a spoiler to the post. I'm fully aware that the Discussions is basically markdown and that has no native support for spoilers, so there's a good possibility that they will just dismiss this suggestion outright.

### Why would a spoiler be useful? 
Well, let's say I'm making a post about coding challenges that a noob can try out. I want them to figure it out for themselves and then post their solution to the thread for other people to see how different people made different solutions. Maybe I even want to post my own solution to the challenge. Well, it'd help if the solutions were able to be hidden until clicked on. This way folk who want to go at the challenge without "cheating" won't accidentally glance at a solution and be influenced.

**Reply by Griatch** [reply]
Markdown also supports HTML tags and you can use the `<details>` tag for this: 

```markdown
Question: 
How do I do this or that? 
<details>
<summary>
Click here for the answer!
</summary>
The answer is ...
</details>
```
----
Question: 
How do I do this or that? 
<details>
<summary>
Click here for the answer!
</summary>
The answer is ...
</details>

**Reply by Griatch** [reply]
I have expanded the first post with some more info about this.

**Reply by kovitikus** [reply]
@Griatch This may be one of those select posts worth pinning, if this is meant to be the first thing people read when they start out here. The markdown is a great addition! Thanks. 💯

**Reply by Griatch** [reply]
Already added this, yes. And this post is already pinned :wink:

**Comment by kovitikus** [comment]
Here's another suggestion for you @Griatch.

I can't seem to restrict who can create a discussion in certain categories, ~~nor can I restrict who can respond to a specific discussion.~~

Seems that locking a discussion will prevent anyone from being able to respond, except for either collaborators or repo owner, depending on your settings. So I suppose this is a way to lock down a discussion so that only specific folk can respond to, although it seems really crude and doesn't allow much nuance. It still doesn't address category restrictions.

![Lock Discussion Option](https://i.imgur.com/Wuc0feI.png)

***

I created a [development blog](https://github.com/kovitikus/hecate/discussions?discussions_q=category%3A%22Hecate+Development+Blog%22) of my own on Hecate and it would be nice if I could restrict others from creating a new discussion in that category, but still allow replies on each individual discussion. This way I could get feedback from other folk, but I don't have rogue individuals posting in the category outside of its purpose.

**Reply by Griatch** [reply]
More fine-grained permissions have already been requested by others to github it seems. Github seems positive to implementing it, but when is unknown. I seems like it would support something like this in that case.
